src/sonic-yang-models/yang-models/sonic-mclag.yang

```diff
@@ -0,0 +1,240 @@
+module sonic-mclag {
+	namespace "http://github.com/Azure/sonic-mclag";
+	prefix smclag;
+	yang-version 1.1;
+
+
+    import ietf-yang-types {
+        prefix yang;
+    }
+
+	import ietf-inet-types {
+		prefix inet;
+	}
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    import sonic-extension {
+        prefix sonic-ext;
+    }
+
+    import sonic-port {
+        prefix prt;
+    }
+
+    import sonic-portchannel {
+        prefix spc;
+    }
+
+    import sonic-vlan {
+        prefix svlan;
+    }
+
+	organization
+		"SONiC";
+
+	contact
+		"SONiC";
+
+	description
+		"SONIC MCLAG";
+
+	revision 2019-10-01 {
+		description
+			"Initial revision.";
+	}
+
+	container sonic-mclag {
+
+        container MCLAG_DOMAIN {
+
+            list MCLAG_DOMAIN_LIST { 
```

Comment:
Wondering if we really need list with max-elements 1, can you explain any potential use-case that might come up in the future with more than one MCLAG domains?

Reply:
this is added for future proof to support multi domain config